### PR TITLE
doc: update for NetBSD 9.2, add GUI Build Instructions

### PR DESCRIPTION
**Pull request by jarolrod**
**For reviewer:** as I suppose few have a NetBSD system available, I wrote a [guide](https://gist.github.com/jarolrod/385dc063bb02c90aea0cbe8a147fc418#file-netbsd-vm-setup-guide-md) to setup a VM for testing purposes.

This attempts to update the NetBSD docs so one can successfully build on the latest release. It also adds instructions to build the GUI.

Additionally, it includes a note and an example on how one could update the gcc version bundled with NetBSD 9.2 and prior to be able to actually compile. This note can be updated with the release of NetBSD 10, as it will package an acceptable gcc version.

Master: [render](https://github.com/bitcoin/bitcoin/blob/master/doc/build-netbsd.md)
PR: [render](https://github.com/bitcoin/bitcoin/blob/d3e9a1c71bef1d730b5820f85e9758af54267ac3/doc/build-netbsd.md)

Related to #20610, but reworked.

### CODE REVIEW

File: doc/build-netbsd.md [hunk 1]
```diff
@@ -1,92 +1,144 @@
...
+### 3. Install Optional Dependencies
 
-### Building Bitcoin Core
+#### Wallet Dependencies
 
-**Important**: Use `gmake` (the non-GNU `make` will exit with an error).
+It is not necessary to build wallet functionality to run bitcoind or the GUI.
 
-#### With descriptor wallet:
+###### Legacy Wallet Support
```
Comment (fanquake) [inline doc/build-netbsd.md:35]: Please leave the descriptor wallet section first.

File: doc/build-netbsd.md [hunk 2]
```diff
@@ -1,92 +1,144 @@
...
+
+`db4` is required to enable support for legacy wallets. Skip if you don't intend to use legacy wallets.
```
Comment (fanquake) [inline doc/build-netbsd.md:37]: > Skip if you don't intend to use legacy wallets.

I think sentences like this (repeated for every dependency) are overly verbose (and somewhat obvious) and could be dropped.

File: doc/build-netbsd.md [hunk 3]
```diff
@@ -1,92 +1,144 @@
...
+
+**Note**: Use `gmake` (the non-GNU `make` will exit with an error).
+
+
+### 1. Configuration
+
+There are many ways to configure Bitcoin Core. Here are a few common examples:
+##### Wallet (BDB + SQlite) Support, No GUI:
+
+This explicitly enables legacy wallet support and disables the GUI. An error will be thrown if `db4` is not installed. If `sqlite3` is installed, then descriptor wallet support will also be built.
```
Comment (fanquake) [inline doc/build-netbsd.md:84]: > An error will be thrown if `db4` is not installed.

I really don't this we need to say things like this in this document.

I would rather drop all these example configuration examples (maybe except for one), and tell users they can look at the output of ./configure --help.

File: doc/build-netbsd.md [hunk 4]
```diff
@@ -1,92 +1,144 @@
...
 ```bash
 ./autogen.sh
-./configure --with-gui=no CPPFLAGS="-I/usr/pkg/include" \
+./configure --without-bdb --with-gui=yes \
+    CPPFLAGS="-I/usr/pkg/include" \
     LDFLAGS="-L/usr/pkg/lib" \
-    BOOST_CPPFLAGS="-I/usr/pkg/include" \
-    BDB_LIBS="-L${BDB_PREFIX}/lib -ldb_cxx-4.8" \
-    BDB_CFLAGS="-I${BDB_PREFIX}/include" \
+    --with-boost-libdir=/usr/pkg/lib \
```
Comment (fanquake) [inline doc/build-netbsd.md:104]: Is using BOOST_CPPFLAGS not sufficient? We no-longer link against any Boost libraries.

File: doc/build-netbsd.md [hunk 5]
```diff
@@ -1,92 +1,144 @@
...
-libevent
-libtool
-pkg-config
-python37
 
+See [dependencies.md](dependencies.md) for a complete overview.
+
+### 2. Clone Bitcoin Repo
+
+Now that `git` and the required dependencies are installed, let's clone the Bitcoin Core repository to a directory. All build scripts and commands will run from this directory.
```
Comment (fanquake) [inline doc/build-netbsd.md:23]:
```suggestion
Clone the Bitcoin Core repository to a directory. All build scripts and commands will run from this directory.
```

**Review by fanquake — COMMENTED** [review]
concept ACK

**Comment by jarolrod** [comment]
updated from d43fc74 to d3e9a1c ([pr25549.01](https://github.com/jarolrod/bitcoin/tree/pr25549.01) -> [pr25549.02](https://github.com/jarolrod/bitcoin/tree/pr25549.02), [diff](https://github.com/jarolrod/bitcoin/compare/pr25549.01..pr25549.02))

Addressed @fanquake comments: thanks for the review, I feel it is now more concise and to the point

File: doc/build-netbsd.md (hunk 4 above)
Comment (jarolrod) [reply doc/build-netbsd.md]: actually i confirmed the only thing we now need is CPPFLAGS

File: doc/build-netbsd.md (hunk 3 above)
Comment (jarolrod) [reply doc/build-netbsd.md]: true, i left the example where the wallet and gui are explicitly disabled.

File: doc/build-netbsd.md [hunk 6]
```diff
@@ -1,92 +1,144 @@
...
     CPPFLAGS="-I/usr/pkg/include" \
     LDFLAGS="-L/usr/pkg/lib" \
-    BOOST_CPPFLAGS="-I/usr/pkg/include" \
+    --with-boost-libdir=/usr/pkg/lib \
     MAKE=gmake
 ```
 
+##### Note on GCC
+
+We currently require GCC 8.1 or greater to compile; NetBSD currently ships with an older version. You should upgrade your GCC and then pass this new version to the configure script.
```
[the remark this answers is not in the document]
Comment (jarolrod) [reply doc/build-netbsd.md]: moved to `1. Install required dependencies`, also removed reference to any exact version of gcc we require. It's only needed to state that netbsd currently ships with a version which is less than is needed.

**Review by aureleoules — COMMENTED** [review]
ACK d3e9a1c71bef1d730b5820f85e9758af54267ac3.
I was able to compile Bitcoin Core (with GUI) on a NetBSD 9.2 virtual machine using jarolrod's guide and followed the build guide without any issue.

**Review by fanquake — APPROVED** [review]
ACK d3e9a1c71bef1d730b5820f85e9758af54267ac3

**Pull request merged**